### PR TITLE
Convert the Remix SPA mode page into a guide for apps without Clerk

### PR DESCRIPTION
**Pull request by anagstef**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1266/references/remix/spa-mode
> - https://clerk.com/docs/pr/1266/quickstarts/remix

### Explanation:
The Remix SPA mode docs page was showcasing how to convert an existing Clerk-Remix project into a SPA mode. Probably, this won't be the case for most people.

### This PR: 

- Creates a guide similar to the Remix quickstart, that will help devs install Clerk into a Remix that uses the SPA mode.

### CODE REVIEW

File: docs/references/remix/spa-mode.mdx [hunk 1]
```diff
@@ -43,6 +66,52 @@ function App() {
   );
 }
 
+export default ClerkApp(App, {
+  publishableKey: 'pk_test_XYZ',
```
Comment (royanger) [inline docs/references/remix/spa-mode.mdx:70]: Does this not read the `CLERK_PUBLISHABLE_KEY` environment variable the way the SSR version does?

**Review by royanger — APPROVED** [review]

**Review by panteliselef — APPROVED** [review]

File: docs/references/remix/spa-mode.mdx (hunk 1 above)
Comment (anagstef) [reply docs/references/remix/spa-mode.mdx:70]: @royanger Currently, no. I will research in a later release, if we can leverage Vite's `import.meta.env`.

Comment (anagstef) [reply docs/references/remix/spa-mode.mdx:70]: Also, this would require users to use the `VITE_` prefixed variables. 

Do you think it would be confusing?

**Review by octoper — APPROVED** [review]
💯

File: docs/references/remix/spa-mode.mdx (hunk 1 above)
Comment (anagstef) [reply docs/references/remix/spa-mode.mdx:70]: @royanger Unfortunately, after investigating a bit, the `import.meta` is not available in CommonJS modules, which we still support for Remix, so it's not an easy win to support this. We will revisit this in the future, maybe when we drop commonjs support in a future major version.

Comment (anagstef) [reply docs/references/remix/spa-mode.mdx]: I decided to change it and provide a similar code snippet as we do in our React quickstart, where we set the `.env` file and then we suggest the users uses the `import.meta` themselves in their `app/root.tsx` file. I think this is a good middleground for now, and we will revisit in the future.

**Pull request merged**